File: Q14.py
arr = ["flower", "flow", "flight"]
res = []
first = arr[0]

for j in range(len(first)):
    c = first[j]
    # print(c)
    flag = True
    for i in arr:
        if len(i) - 1 >= j:
            if c != i[j]:
                flag = False
                break
        else:
            flag = False
            break
    if flag:
        res.append(c)
    else:
        break

print("".join(res))

File: Q14.js
let arr = ["flower", "flow", "flight"];
let first = arr[0];
let res = [];

for (let j = 0; j < first.length; j++) {
    let c = first[j];
    let flag = true;
    for (let i of arr) {
        if (i.length - 1 >= j) {
            if (c != i[j]) {
                flag = false;
                break;
            }
        } else {
            flag = false;
            break;
        }
    }
    if (flag) {
        res.push(c);
    } else {
        break;
    }
}

console.log(res.join(""));